desktop/renderer/index.js

```diff
@@ -52,59 +48,33 @@ class Keybase extends Component {
     // Used by material-ui widgets.
     injectTapEventPlugin()
 
-    // For remote window components
-    ipcMain.removeAllListeners('dispatchAction')
-    ipcMain.removeAllListeners('stateChange')
-    ipcMain.removeAllListeners('subscribeStore')
-
-    ipcMain.on('dispatchAction', (event, action) => {
-      // we MUST clone this else we'll run into issues with redux. See https://github.com/rackt/redux/issues/830
-      // This is because we get a remote proxy object, instead of a normal object
-      // Using _.cloneDeep() creates a non-plain object
-      setImmediate(() => store.dispatch(_.merge({}, action)))
-    })
-
-    ipcMain.on('subscribeStore', event => {
-      const sender = event.sender // cache this since this is actually a sync-rpc call...
-
-      // Keep track of the last state sent so we can make the diffs.
-      let oldState = {}
-      const getStore = () => {
-        const newState = store.getState()
-        const diffState = shallowDiff(oldState, newState) || {}
-        oldState = newState
-        return diffState
-      }
+    this.setupDispatchAction()
+    this.setupStoreSubscriptions()
 
-      if (!sender.isDestroyed()) {
-        sender.send('stateChange', getStore())
-      }
+    // Handle notifications from the service
+    store.dispatch(listenForNotifications())
 
-      let unsubscribe = store.subscribe(() => {
-        if (sender.isDestroyed()) {
-          if (unsubscribe) {
-            unsubscribe()
-            unsubscribe = null
-          }
-          return
-        }
+    // Handle logUi.log
+    ListenLogUi()
+  }
 
-        const diffState = getStore()
-        if (Object.keys(diffState).length !== 0) {
-          if (!sender.isDestroyed()) { // We need this extra check due to timing issues
-            sender.send('stateChange', diffState)
-          }
+  setupDispatchAction () {
+    ipcRenderer.on('dispatchAction', (event, action) => {
+      // // we MUST convert this else we'll run into issues with redux. See https://github.com/rackt/redux/issues/830
+      // // This is because we get a remote proxy object, instead of a normal object. Using _.cloneDeep() creates a non-plain object
+      setImmediate(() => {
+        try {
+          store.dispatch(_.merge({}, action))
+        } catch (_) {
         }
       })
     })
+  }
 
-    ipcRenderer.send('remoteStoreReady')
-
-    // Handle notifications from the service
-    store.dispatch(listenForNotifications())
-
-    // Handle logUi.log
-    ListenLogUi()
+  setupStoreSubscriptions () {
+    store.subscribe(() => {
+      ipcRenderer.send('stateChange', store.getState())
```

Comment:
So we're still sending the whole state every time the state changes to the main thread? Is the a remote proxy object or is this serialized?